test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,225 @@
+		It("filters devices by label with contains operator", func() {
+			deviceName := extractDeviceName()
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.labels contains region=eu-west-1")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+
+		It("filters devices by multiple field selectors", func() {
+			deviceName := extractDeviceName()
+			startTimestamp, _ := generateTimestamps()
+			out, err := harness.CLI("get", "devices", "--field-selector",
+				fmt.Sprintf("metadata.labels contains region=eu-west-1,metadata.creationTimestamp>=%s", startTimestamp))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring(deviceName))
+		})
+
+		It("excludes devices by name", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name!=device1-name")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).ToNot(ContainSubstring("device1-name"))
+		})
+	})
+
+	Context("Negative Tests", Label("77948"), func() {
+		It("returns an error for an invalid field selector", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "invalid.field")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring(unknownSelector))
+		})
+		It("returns an error for an unsupported field selector", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "unsupported.field")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring(unknownSelector))
+		})
+
+		It("returns an error for an invalid operator", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name@=device1-name")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring(unknownSelector))
+		})
+
+		It("returns an error for an incorrect field type", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name>10")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring("unsupported for type string"))
+		})
+
+		It("returns an error for an invalid timestamp format", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "'metadata.creationTimestamp>=2024-01-01'")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring("value must be a number or a valid time in RFC3339 format"))
+		})
+
+		It("returns an error for no field selector (an empty string field selector is valid)", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring("flag needs an argument"))
+		})
+
+		It("returns an error for multiple errors in the field selector", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "invalid.field,metadata.name@=device1-name")
+			Expect(err).To(HaveOccurred())
+			Expect(out).To(ContainSubstring(unknownSelector))
+		})
+
+		It("Deleting resources", func() {
```

Comment:
Maybe this cleaning could be in an AfterSuite func, since with "It" it will be reported as a test case